Doctests confuse multi-paragraph footnote with indented code block

I tried this code:

```rust
/// Example
///
/// Footnote with multiple paragraphs[^multiple]
///
/// [^multiple]:
///     One
///
///     Two
///
///     Three
pub fn add(left: u64, right: u64) -> u64 {
    left + right
}
```

This is a footnote with multiple paragraphs. It renders correctly in the documentation:

<img width="303" alt="Image" src="https://github.com/user-attachments/assets/6d0faeaa-632f-4d43-b340-13cc42a22247" />

However, running as a doctest fails:

```
running 1 test
test src/lib.rs - add (line 8) ... FAILED

failures:

---- src/lib.rs - add (line 8) stdout ----
error: expected `;`, found `Three`
 --> src/lib.rs:9:4
  |
3 | Two
  |    ^ help: add `;` here
4 |
5 | Three
  | ----- unexpected token

error[E0425]: cannot find value `Three` in this scope
 --> src/lib.rs:11:1
  |
5 | Three
  | ^^^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.

failures:
    src/lib.rs - add (line 8)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s
```

Somehow the doctest collector seems to not understand that there are footnotes. I would expect this to be ignored.

### Meta

```
rustdoc 1.87.0-nightly (a2e63569f 2025-03-26)
```

--- a/src/librustdoc/html/markdown.rs
+++ b/src/librustdoc/html/markdown.rs
@@ -721,7 +721,7 @@ pub(crate) fn find_codes<T: doctest::DocTestVisitor>(
     extra_info: Option<&ExtraInfo<'_>>,
     include_non_rust: bool,
 ) {
-    let mut parser = Parser::new(doc).into_offset_iter();
+    let mut parser = Parser::new_ext(doc, main_body_opts()).into_offset_iter();
     let mut prev_offset = 0;
     let mut nb_lines = 0;
     let mut register_header = None;

--- a/tests/rustdoc-ui/multi-par-footnote.rs
+++ b/tests/rustdoc-ui/multi-par-footnote.rs
@@ -1,0 +1,18 @@
+//@ check-pass
+//@ compile-flags:--test
+//@ normalize-stdout: "finished in \d+\.\d+s" -> "finished in $$TIME"
+// Regression test for #139064.
+
+/// Example
+///
+/// Footnote with multiple paragraphs[^multiple]
+///
+/// [^multiple]:
+///     One
+///
+///     Two
+///
+///     Three
+pub fn add(left: u64, right: u64) -> u64 {
+    left + right
+}

--- a/tests/rustdoc-ui/multi-par-footnote.stdout
+++ b/tests/rustdoc-ui/multi-par-footnote.stdout
@@ -1,0 +1,5 @@
+
+running 0 tests
+
+test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in $TIME
+

